runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobServerDriver.java

```diff
@@ -139,6 +145,10 @@ private InMemoryJobService createJobService() throws IOException {
             throw new RuntimeException(exn);
           }
         },
+        configuration.cleanArtifactsPerJob ?
+            (String stagingSessionToken) ->
+                artifactStagingService.getService().removeArtifacts(stagingSessionToken)
+            : null,
```

Comment:
This will throw NullPointer on accept call if configuration.cleanArtifactsPerJob is false.

I was suggesting: 
```
(String stagingSessionToken) -> if(configuration.cleanArtifactsPerJob) artifactStagingService.getService().removeArtifacts(stagingSessionToken)
```

Reply:
> This will throw NullPointer on accept call if configuration.cleanArtifactsPerJob is false.

I don't think that's true, as written? [I check `cleanupJobFn != null` in the only place it's referenced in `InMemoryJobService`](https://github.com/apache/beam/pull/5958/files#diff-d4d147290be404469ebc926e593a187eR185).

Or maybe I'm missing what you mean?

Anyway, if you think this way is cleaner I'm almost fine with it, lmk!